frontend/src/components/core/StreamlitDialog/StreamlitDialog.tsx

```diff
@@ -108,6 +115,69 @@ export function StreamlitDialog(dialogProps: DialogProps): ReactNode {
   }
 }
 
+interface EmbedProps {
+  type: DialogType.EMBED
+
+  /** Callback to close the dialog */
+  onClose: PlainEventHandler
+
+  embedURL: string
+}
+
+/**  */
+function embedDialog(props: EmbedProps): ReactElement {
+  const markdownStyle: CSSProperties = {
+    overflowY: "auto",
+    overflowX: "hidden",
+    maxHeight: "35vh",
+  }
+
+  const initialMessage =
+    "To embed this app, please place the following HTML snippet somewhere on your website:"
+
+  const codeText = `<iframe  \n \t width="100%" \n \t height="500px" \n \t frameBorder="0" \n \t src="${props.embedURL}" \n ></iframe>`
+
+  // Markdown New line is 2 spaces + \n
+  const newLineMarkdown = "  \n"
+  const customizationOptionsMd = `${
+    "" +
+    "There are multiple options to customize embed (they should be placed as embed query params):"
+  }${newLineMarkdown}* \`?embed=noColoredBar\` makes the colored bar at the top disappear.${newLineMarkdown}* \`?embed=noToolBar\` makes the toolbar disappear.${newLineMarkdown}* \`?embed=noPadding\` removes the extra padding at the top and bottom of the app.${newLineMarkdown}* \`?embed=noScroll\` disables scrolling.${newLineMarkdown}* \`?embed=noFooter\` removes "Made with Streamlit" footer.${newLineMarkdown}${newLineMarkdown}Embed params can be merged in the following way:${newLineMarkdown}\`?embed=noColoredBar&embed=noToolBar\`.${newLineMarkdown}Param \`?embed=true\` will always override other options, so do not use it when trying to customize embed params.${newLineMarkdown}`
+
+  return (
+    <Modal isOpen onClose={props.onClose}>
+      <ModalHeader>How to embed this app?</ModalHeader>
+      <ModalBody>
+        {initialMessage}
+        <StyledCopyButtonInEmbedModalContainer>
+          <CopyButton
+            alwaysVisible={true}
+            text={codeText}
+            onClick={() => {
+              navigator.clipboard.writeText(codeText).then(r => {
```

Comment:
shouldn't we use async/await?

Reply:
I don't think so, I don't see a purpose. Is there a reason we should use it here?